src/phases/pokemon-phase.ts

```diff
@@ -4,21 +4,23 @@ import type Pokemon from "#app/field/pokemon";
 import { FieldPhase } from "./field-phase";
 
 export abstract class PokemonPhase extends FieldPhase {
-  protected battlerIndex: BattlerIndex | number;
+  protected battlerIndex: BattlerIndex;
```

Comment:
```suggestion
  protected battlerIndex: BattlerIndex | number;
```
This is `| number` because it (insanely) assumes that this is actually a pokemon ID and not a `BattlerIndex` if the number is greater than the possible `BattlerIndex` values. You could probably rename it to `battlerIndexOrId` for now so at least it's clear that's what is going on.

Reply:
this... is even worse than the bleeping "oops 0 PID means bork" issues with transferring items